day_10/task_2/tests/pipe_maze_tests.cpp

```diff
@@ -131,4 +131,42 @@ std::vector<MazeScopeParams> GenerateMazeScopeParams()
 
 INSTANTIATE_TEST_SUITE_P(IsInMazeScope,
                          MazeScopeTests,
-                         testing::ValuesIn(GenerateMazeScopeParams()));
\ No newline at end of file
+                         testing::ValuesIn(GenerateMazeScopeParams()));
+
+
+TEST_P(MazeScopeTests, GivenMazePoint_WhenColorNeighborForPoint_ExpectColorEightPoint)
+{
+    // "- L | F 7",
+    // "7 S - 7 |",
+    // "L | 7 | |",
+    //  "- L - J |",
+    //  "L | -J F"};
+
+    auto [color_right, color_top_right, color_down_right, color_top, color_down,
+          color_left, color_top_left, color_down_left] = GetParam();
+
+    maze.color_neighbor(MazePoint{1, 1});
+    ASSERT_EQ(check_state_at(MazePoint{1, 2}), color_right);
+    ASSERT_EQ(check_state_at(MazePoint{0, 2}), color_top_right);
+    ASSERT_EQ(check_state_at(MazePoint{2, 2}), color_down_right);
+
+    ASSERT_EQ(check_state_at(MazePoint{0, 1}), color_top);
+    ASSERT_EQ(check_state_at(MazePoint{2, 1}), color_down);
+
+    ASSERT_EQ(check_state_at(MazePoint{0, 0}), color_top_left);
+    ASSERT_EQ(check_state_at(MazePoint{1, 0}), color_left);
+    ASSERT_EQ(check_state_at(MazePoint{2, 0}), color_down_left);
+}
+
+using MazeNeighborColorParams = std::tuple<MazePoint, bool>;
```

Comment:
This type doesn't fit to the values in the 165 line.
```suggestion
using ColorTopLeft = MazePoint;
using ColorLeft = MazePoint;
using ColorBottomLeft = MazePoint;
using ColorBottomMid = MazePoint;
using ColorBottomRight = MazePoint;
using ColorRight = MazePoint;
using ColorTopRight = MazePoint;
using ColorTopMid = MazePoint;
using MazeNeighborColorParams = std::tuple<ColorTopLeft, ColorLeft, ColorBottomLeft, ColorBottomMid, ColorBottomRight, ColorRight, ColorTopRight, ColorTopMid>;
```

Reply:
Please notice that the order of colors is counter clockwise (could be optionally clockwise as well - this is arbitrary decision) and hence it is easier to imagine the test case for the reader

Reply:
done